relaynet-cogrpc-jvm/src/test/java/tech/relaycorp/relaynet/cogrpc/client/CogRPCClientTest.kt

```diff
@@ -0,0 +1,97 @@
+package tech.relaycorp.relaynet.cogrpc.client
+
+import io.grpc.BindableService
+import io.grpc.internal.testing.StreamRecorder
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import tech.relaycorp.relaynet.cogrpc.CargoDeliveryAck
+import tech.relaycorp.relaynet.cogrpc.CogRPC
+import tech.relaycorp.relaynet.cogrpc.test.MockCogRPCServerService
+import tech.relaycorp.relaynet.cogrpc.test.TestCogRPCServer
+import tech.relaycorp.relaynet.cogrpc.test.Wait.waitForNotNull
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+internal class CogRPCClientTest {
```

Comment:
I'd add tests for the following cases (copy/pasting the tests from the JS implementation):

- Main/constructor
  - `gRPC client must be initialized with specified server address`
  - `TSL should be used by default`
  - `TSL can be disabled`
- deliverCargo
  - `Deadline should be set to 5 seconds`
  - `Unknown delivery ids should end the call and throw an error` <- I'm happy to ignore unknown ids instead, but whatever we do, I think we should have a test for that so we can make sure it'll be handled properly.
  - `Connection should be closed when all delivery requests have been acknowledged`
  - `Error should be thrown when connection ends with outstanding acknowledgements`
- collectCargo
  - `Deadline should be set to 5 seconds`
  - `Error should be thrown when server ends call with PERMISSION_DENIED status`